server/utils.js

```diff
@@ -22,3 +22,14 @@ exports.geoDist = (lat1, lon1, lat2, lon2) => {
 exports.flatMap = (listVals, defaultVal = null) => {
   return listVals.length > 0 ? listVals[0] : defaultVal;
 };
+
+// Split an array into sub-arrays of up to length chunkSize.
+// Returns an arry of the sub-arrays.
+exports.chunk = (array, chunkSize) => {
+  const res = [];
+  let i = 0;
+  while (i < array.length) {
+    res.push(array.slice(i, (i += chunkSize)));
```

Comment:
Try to avoid the trick like using an expression as a parameter.
It would be confusing as the expression has side effect.
It could also be dangerous when you are not sure which of the two variables would be evaluated first:
Image i += chunkSize has been evaluated first, then the first i would become the same value as the second one.

Split this to two lines:
res.push(array.slice(i, (i + chunkSize))); 
i += chunkSize;

Or use a for loop would be more straightforward.

Reply:
To clairy, I do not mean not use expression as parameter, but expression which would has side effect.